### PR TITLE
Task/Chore: Add linting for Typescript files

### PR DESCRIPTION
**Pull request by andre-meyer**
# Description

Typescript files weren't previously linted by the pre-commit hook. This PR adds linting for ts files and fixed most problems that required manual work as well as all automatic fixes and all errors.

The remaining linting warnings are:
image

Here's some changes I had to make:
```
/Users/andremeyer/Documents/Projects/safe-app-gnosis-protocol/src/utils/finalForm.ts
  5:55  error    Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
- If you want a type meaning "any value", you probably want `unknown` instead  @typescript-eslint/ban-types
```

```
/Users/andremeyer/Documents/Projects/safe-app-gnosis-protocol/src/components/pages/deploy/DeployPage.stories.tsx
  93:39  error    Unexpected empty async arrow function  @typescript-eslint/no-empty-function
```

Those are the kind of changes I made immediately, the remaining ones I don't feel safe just changing. Please either fix them whenever or ignore them for now @alfetopito 

# To Test
- `npm run lint -- --fix` runs without error

# Background

New dependency `@typescript-eslint/parser`

### CODE REVIEW

File: src/components/basic/display/MarketPrice/index.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ export interface Props {
   setError?: (msg?: string) => void;
 }
 
-function component(props: Props): JSX.Element {
+function UMarketPrice(props: Props): JSX.Element {
```
Comment (alfetopito) [inline src/components/basic/display/MarketPrice/index.tsx:15]: Why the `U` prefix?

File: src/components/pages/deploy/DeployForm.tsx [hunk 2]
```diff
@@ -283,4 +284,8 @@ const component: React.FC = ({ children }) => {
   );
 };
 
-export const DeployForm: typeof component = memo(component);
+UDeployForm.propTypes = {
+  children: PropTypes.node.isRequired,
+};
```
Comment (alfetopito) [inline src/components/pages/deploy/DeployForm.tsx:289]: In that case I'd rather have it as explicit Props interface/type rather than using `propTypes`

**Review by alfetopito — CHANGES_REQUESTED** [review]
1 question and 1 change about propTypes

File: src/components/basic/display/MarketPrice/index.tsx (hunk 1 above)
Comment (andre-meyer) [reply src/components/basic/display/MarketPrice/index.tsx:15]: U stands for Unsafe or Unmemorized. `component` is a name that causes a ton of issues with typescript and react.

File: src/components/pages/deploy/DeployForm.tsx (hunk 2 above)
Comment (andre-meyer) [reply src/components/pages/deploy/DeployForm.tsx:289]: Yup! I figured this out later but didn't change it everywhere. Will change.

File: src/components/basic/display/MarketPrice/index.tsx (hunk 1 above)
Comment (alfetopito) [reply src/components/basic/display/MarketPrice/index.tsx:15]: As discussed over slack, I'll remove the intermediate component and apply `memo` directly.

**Pull request merged**